pages/api/dpu/repos.ts

```diff
@@ -1,25 +1,48 @@
 import { NextApiRequest, NextApiResponse } from "next";
-import { getUserRepositoriesByTopic } from "../../../utils/db/repos";
 import { RepoIdentifier } from "../../../types/repository";
+import { getUserRepositoriesByTopic } from "../../../utils/db/repos";
+import { getUserIdByTopicName } from "../../../utils/db/users";
+import rudderStackEvents from '../events';
 
 const reposHandler = async (req: NextApiRequest, res: NextApiResponse) => {
-    const { topicId, provider } = req.query;
-    if (!topicId || !provider 
-        || Array.isArray(topicId) || Array.isArray(provider)
-        || topicId.length === 0 || provider.length === 0) {
-        res.status(400).json({ error: 'Invalid get request body' });
+	const { topicId, provider } = req.query;
+	
+	const event_properties = {
+		topic_name: topicId as string,
+		repo_provider: provider as string,
+	};
+
+	if (!topicId || !provider 
+		|| Array.isArray(topicId) || Array.isArray(provider)
+		|| topicId.length === 0 || provider.length === 0) {
+		res.status(400).json({ error: 'Invalid get request body' });
+		const eventProperties = { ...event_properties, response_status: 400 };
+		rudderStackEvents.track("absent", "", 'dpu-repos', { type: 'invalid-body', eventStatusFlag: 0, eventProperties });
```

Comment:
Across all files, can we have the same type for a particular error, like 'HTTP-400' instead of 'invalid-body' which might change in other files

Reply:
For other type of errors, the types right now look good

Reply:
sure